ksqldb-common/src/main/java/io/confluent/ksql/function/types/ParamTypes.java

```diff
@@ -38,6 +38,7 @@ private ParamTypes() {
   public static final LongType LONG = LongType.INSTANCE;
   public static final ParamType DECIMAL = DecimalType.INSTANCE;
   public static final TimestampType TIMESTAMP = TimestampType.INSTANCE;
+  public static final KsqlLambdaType LAMBDALITERAL = KsqlLambdaType.of(null, null);
```

Comment:
Looking at this out of context doesn't really make sense to me. Creating a lambda type with no input/output type seems like a bug (and using `null` to proxy this makes me feel even more nervous, in fact I'd probably recommend that we `requireNonNull` for these in the constructor). I'll try to look throughout at where this is used and comment with some more details later in the PR, but I suspect we should follow the pattern of `MapType` and `ArrayType` which are complex types (they don't have corresponding literals in this list)